### PR TITLE
Adds support for Provider mode deployment in installation flow

### PR DESCRIPTION
**Pull request by bipuladh**
Screenshots: 
Screenshot from 2024-03-19 14-28-31
Screenshot from 2024-03-15 14-07-27
Screenshot from 2024-03-15 14-07-39
Screenshot from 2024-03-19 14-28-24

### CODE REVIEW

File: locales/en/plugin__odf-console.json [hunk 1]
```diff
@@ -756,6 +756,7 @@
   "Files should be named private and public followed by compatible extensions": "Files should be named private and public followed by compatible extensions",
   "Deploys MultiCloud Object Gateway without block and file services.": "Deploys MultiCloud Object Gateway without block and file services.",
   "Deploys Data Foundation with block, shared fileSystem and object services.": "Deploys Data Foundation with block, shared fileSystem and object services.",
+  "Deploys Data Foundation as a provider cluster": "Deploys Data Foundation as a provider cluster",
```
Comment (bipuladh) [inline locales/en/plugin__odf-console.json:759]: Needs input from doc team.

**Review by SanjalKatiyar — COMMENTED** [review]
LGTM.

Couple of doubts:
1. Don't we want to disable `Connect an external storage platform` radio when deployment mode is provider ?
2. Don't we want to show NFS checkbox for provider ?

Minor nit:
1. In review step, we can hide DR related info if we are not showing `Data protection` step.

**Comment by bipuladh** [comment]
@SanjalKatiyar updated as requested

**Pull request merged**